include/display.h

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = (1 << 0),
+};
+
+/**
+ * @struct display_capabilities
+ * @brief Structure holding display capabilities
+ *
+ * @var display_capabilities::x_resolution
+ * Display resolution in the X direction
+ *
+ * @var display_capabilities::y_resolution
+ * Display resolution in the Y direction
+ *
+ * @var display_capabilities::supported_pixel_formats
+ * Bitwise or of pixel formats supported by the display
+ */
+struct display_capabilities {
+	u16_t x_resolution;
+	u16_t y_resolution;
+	enum display_pixel_format supported_pixel_formats;
+};
+
+/**
+ * @typedef display_on_api
+ * @brief Callback API to turn display on
+ * See display_on() for argument description
+ */
+typedef int (*display_on_api)(struct device *dev);
+
+/**
+ * @typedef display_off_api
+ * @brief Callback API to turn display off
+ * See display_off() for argument description
+ */
+typedef int (*display_off_api)(struct device *dev);
+
+/**
+ * @typedef display_write_bitmap_api
+ * @brief Callback API for writing bitmap
+ * See display_write_bitmap() for argument description
+ */
+typedef int (*display_write_bitmap_api)(
+	const struct device *dev, const u16_t x, const u16_t y, const u16_t w,
```

Comment:
Have you considered support for images with stride?

Reply:
No the interface was designed with embedded LCD displays in mind. Most of the time these devices require a window where they operate on and will write pixels consecutively in to this window. So if a stride is used the window needs to repositioned for every stride that takes place.

An option would be to extend  the interface to take a pointer to an array of bitmaps:
```
struct bitmap {
void* data
u16_t x;
u16_t y;
u16_t w;
u16_t h;
void *data;
};
typedef int (*display_write_bitmap_api)(
	const struct device *dev, cosnt struct bitmap  *bitmaps, u32_t nbr_of_bitmaps);
```